client/data/marketplace/use-es-query.ts

```diff
@@ -80,6 +83,11 @@ const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
 			icon: createIconUrl( hit.slug, hit.plugin.icons ),
 			railcar,
 		};
+
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		plugin.variations = getPreinstalledPremiumPluginsVariations( plugin );
```

Comment:
Non-blocking questions: Why do you need to disable this eslint rule? Why the `variations` property is not defined along with the rest of properties above?

Reply:
> Why do you need to disable this eslint rule?

Because I can't use `@ts-ignore` otherwise. 😄 
(Calypso is full of those two lines, just try searching for it!)

I'm not good with TS, and I'm unsure how to fix this error without adding types all over the place. 🤔 
Would love to hear suggestions, though!
```
Type '{}' is missing the following properties from type '{ monthly: { product_slug: string; }; yearly: { product_slug: string; }; }': monthly, yearly
ts(2739)
```

> Why the variations property is not defined along with the rest of properties above?

It uses `plugin` itself, to check for `plugin.slug` and `plugin.variations`.
I guess I could pass down those two properties specifically, but I think this way makes the code slightly tighter when used with different data formats (ES and regular REST  API).

Reply:
> I'm unsure how to fix this error without adding types all over the place

I think this would work but feel free to keep the `@ts-ignore` if you want!

```diff
diff --git a/client/data/marketplace/use-es-query.ts b/client/data/marketplace/use-es-query.ts
index 799a90a25d..a162d072a1 100644
--- a/client/data/marketplace/use-es-query.ts
+++ b/client/data/marketplace/use-es-query.ts
@@ -84,8 +84,6 @@ const mapIndexResultsToPluginData = ( results: ESHits ): Plugin[] => {
 			railcar,
 		};
 
-		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-		// @ts-ignore
 		plugin.variations = getPreinstalledPremiumPluginsVariations( plugin );
 
 		return plugin;
diff --git a/client/lib/plugins/utils.js b/client/lib/plugins/utils.js
index 20b4ba2bb7..7898477c57 100644
--- a/client/lib/plugins/utils.js
+++ b/client/lib/plugins/utils.js
@@ -377,12 +377,20 @@ export const getManageConnectionHref = ( siteSlug ) => {
 		: `/settings/manage-connection/${ siteSlug }`;
 };
 
+/**
+ * @typedef {object} PluginVariations
+ * @property {object} monthly
+ * @property {string} monthly.product_slug
+ * @property {object} yearly
+ * @property {string} yearly.product_slug
+ */
+
 /**
  * Some plugins can be preinstalled on WPCOM and available as standalone on WPORG,
  * but require a paid upgrade to function.
  *
  * @param {object} plugin
- * @returns {object}
+ * @returns {PluginVariations}
  */
 export function getPreinstalledPremiumPluginsVariations( plugin ) {
 	if ( ! PREINSTALLED_PREMIUM_PLUGINS[ plugin.slug ] || !! plugin.variations ) {
```
> It uses plugin itself

Oh right, didn't notice that!